### ISSUE TITLE
Combobox headers not associated with items for screenreader

### ISSUE THREAD

**Issue by hoovercj** [microsoft/fluentui#14906]
### Environment Information

- **Package version(s)**: 7.134.1
- **Browser and OS versions**: Windows 10, Edge 85

### Describe the issue:
Narrator cannot associate Combobox options with their headings like it can for Dropdown options.

In the Dropdown control, when a heading is present, the heading the following items are wrapped in a group which is labelled by the heading item. This pattern is not followed in a Combobox control.

### Please provide a reproduction of the issue in a codepen:

[Combobox control ](https://codepen.io/hoovercj/pen/mdPqxML?editable=true%3Dhttps%3A%2F%2Fdeveloper.microsoft.com%2Fen-us%2Ffluentui)(No association between options and headers)

[Dropdown control](https://codepen.io/hoovercj/pen/zYqPWEK?editable=true%3Dhttps%3A%2F%2Fdeveloper.microsoft.com%2Fen-us%2Ffluentui) (There **is** an association between options and headers):

#### Actual behavior:

When navigation the options of the combobox control, Narrator does not provide information about the current section/heading.

#### Expected behavior:

When navigating through the options, Narrator should read the option and the section/heading just as the dropdown control is doing, to make the user away where he/she is.

### Documentation describing expected behavior

https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html

**Comment by hoovercj** [comment]
I can see that [Dropdown.base.tsx](https://github.com/microsoft/fluentui/blob/master/packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx#L601) processes the list of items to build the groups in its `_onRenderList` function, whereas [ComboBox](https://github.com/microsoft/fluentui/blob/master/packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx#L1318) simply renders each item directly.

**Comment by paulgildea** [comment]
@hoovercj Thanks for the detailed analysis! This is super helpful for the team!

@smhigley FYI

**Comment by miroslavstastny** [comment]
@kolaps33 FYI

**Comment by kolaps33** [comment]
took look quick look on html and agree with @hoovercj 

Looks like to reach narration of the "First heading", then items should be wrapped with div element which has role=group and 'heading' labelling.  

![combobox_listbox_dropdown_listbox](https://user-images.githubusercontent.com/22620150/93371042-fa15e600-f851-11ea-8298-e0436271b81f.png)

**Comment by hoovercj** [comment]
@joschect / @smhigley  -- I'm curious what sort of expectations we should have when we open accessibility bugs on UI fabric based on trusted vendor accessibility testing against our own product. We have 90 day SLAs for our accessibility bugs which we're able to extend a bit by marking them as external and creating bugs on the correct teams (such as this one), but if they aren't ever fixed then we need to work around them.

Can you advise us on how we should treat Fluent UI accessibility bugs going forward to make sure our expectations are appropriate?

**Comment by smhigley** [comment]
Reopening w/ the following proposed fix:

We'll need to modify the DOM for Combobox slightly to insert a `<div role="group">` wrapper around each group of options under a header, but I discussed it w/ @JustSlone, and that should be OK to add in V8. We should essentially do the same thing as Dropdown's `_onRenderList` in Combobox's `_onRenderList`.
